custom_components/google_home/api.py

```diff
@@ -198,7 +201,7 @@ async def update_google_devices_information(self):
 
         coordinator_data = await gather(
             *[
-                self.get_alarms_and_timers(device)
+                self.get_alarms_and_timers(device, device.ip_address)
```

Comment:
Why explicitly passing in `ip_address`?

Reply:
Because `device.ip_address` is `Optional[str]` although we checked here that it's not `None`. In this function we will need `str` to construct url so this is linting time check for that.

Another option is explicitly checking again in `get_alarms_and_timers` that `device.ip_address` is not `None` but that will be runtime check.

Reply:
Can we just remove `Optional` from `GoogleHomeDevice`'s `ip_address` field and make it mandatory with a fallback to `None`?

Reply:
`Optional` is returned from `glocaltokens`. I think there can be a case when ip address is not found in Foyer API.